libcontainer/cgroups/file.go

```diff
@@ -57,6 +57,40 @@ func WriteFile(dir, file, data string) error {
 	return nil
 }
 
+// WriteFileByLine is the same as WriteFile, except if data contains newlines,
+// is is written line by line.
+func WriteFileByLine(dir, file, data string) error {
+	i := strings.Index(data, "\n")
+	if i == -1 {
```

Comment:
I think we can remove this special block.
Others LGTM.

Reply:
@lifubang theoretically, yes. Practically, I added it here because I was overly cautious to not break anything.

With this block in place, if the input does not contain newlines, we use the very same function as we used before. In other words, unless there's a newline in `data`, this commit changes nothing (and it is easy to validate).

It is also easier to read this way (when you debug it, and write a single line, which is the most common scenario, you don't have to read through the code below).